### PR TITLE
Catch and forward kobo http errors

### PR DESCRIPTION
**Pull request by echaidemenos**
This PR resolves https://github.com/WFP-VAM/prism-app/issues/486

Adds a utility function which forwards HTTP errors, with the option to "conceal" some statuses as 500

### CODE REVIEW

File: api/docker-compose.test.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ services:
       - POSTGRES_DB=postgres
       - POSTGRES_PORT=5432
       - POSTGRES_PASSWORD=!ChangeMe!
+      - KOBO_PW=test
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply api/docker-compose.test.yml:14]: KOBO_USERNAME already has a default set in the first docker-compose file so it does not need one here actually

File: api/app/utils.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import requests
+from fastapi import HTTPException
+
+
+def forward_http_error(resp: requests.Response, excluded_codes: list[int]) -> None:
+    try:
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        code = exc.response.status_code
```
Comment (ericboucher) [inline api/app/utils.py:11]: Can't we use the resp object to get this? You should be able to use resp.status_code directly I think.
See how we handled this in this other repo [tap-github - validate_response](https://github.com/MeltanoLabs/tap-github/blob/d9ddf8464731c21eed466c603883551d59e19aaf/tap_github/client.py#L167)

File: api/app/kobo.py [hunk 3]
```diff
@@ -175,7 +174,9 @@ def get_responses_from_kobo(
     else:
         resp = requests.get(form_url, auth=auth)
 
-    resp.raise_for_status()
+    excluded_codes = [403]
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply api/app/kobo.py:179]: yes you are right. it was typo thanks

File: api/app/utils.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+def forward_http_error(resp: requests.Response, excluded_codes: list[int]) -> None:
+    if not resp.status_code > 399:
+        return
+
+    if excluded_codes and resp.status_code in excluded_codes:
+        raise HTTPException(status_code=500, detail="An internal error occurred.")
+    else:
+        obj = resp.json()
+        detail = obj.get("detail", "Unknown error")
```
Comment (ericboucher) [inline api/app/utils.py:15]: Do we have a way to forward more information? Ideally, we would want to make the API error message a bit more useful when possible. What do you think?

**Review by ericboucher — APPROVED** [review]

**Pull request merged**